src/rules/no-extraneous-dependencies.js

```diff
@@ -184,8 +184,15 @@ function reportIfMissing(context, deps, depsOptions, node, name) {
   // test the real name from the resolved package.json
   // if not aliased imports (alias/react for example), importPackageName can be  misinterpreted
   const realPackageName = getModuleRealName(resolved);
+
+  if(!realPackageName){
+    throw new Error(`cant find real package name for import ${name}`);
+  }
+
```

Comment:
instead of defaulting to an empty package name (which is what line 133 effectively did), throw an error when the package name is not found. This will make it easier to find bugs like this in the future.